tests/test_2fa.py

```diff
@@ -1,16 +1,29 @@
+import random
 import secrets
+import string
+from typing import TypeAlias
 
 import pyotp
+import pytest
 from auth_helper import register_user_2fa
 from flask.testing import FlaskClient
 
+LoginData: TypeAlias = dict[str, str]
+
+@pytest.fixture()
+def nonce(len: int = 7) -> str:
+  return ''.join(random.choices(string.ascii_lowercase +  # noqa: S311
+                             string.digits, k=len))
+
+@pytest.fixture()
+def login_data(nonce: str) -> LoginData:
+  return {"username": "test_user" + nonce, "password": "SecurePassword123!"}
```

Comment:
For readability, let's add an underscore between the static name part & the nonce.

```python
"test_user_" + nonce
```

Reply:
addressed by [89fc892](https://github.com/scidsg/hushline/pull/460/commits/89fc892b238cacd35022fa4ab2f19235c2576f1e)